src/rgw/rgw_common.h

```diff
@@ -690,6 +694,43 @@ void decode_json_obj(rgw_placement_rule& v, JSONObj *obj);
 inline std::ostream& operator<<(std::ostream& out, const rgw_placement_rule& rule) {
   return out << rule.to_str();
 }
+
+class RateLimiter;
+struct RGWQoSInfo {
+  int64_t max_write_ops;
```

Comment:
nit:
```suggestion
  int64_t max_write_ops = 0;
```
and remove ctor